### ISSUE TITLE
Website tells me that CSP has 'unsafe-inline' inside style-src, but it doesn't

### ISSUE THREAD

**Issue by leonklingele** [mozilla/http-observatory-website#16]
My CSP: `Content-Security-Policy: default-src 'none'; script-src 'self' https://some-url; object-src 'none'; style-src 'self' data:; img-src 'self' data: https://some-url; media-src 'self'; frame-src 'none'; font-src 'self'; connect-src 'self'`

The test reports `Content Security Policy (CSP) implemented with 'unsafe-inline' inside style-src`

**Comment by marumari** [comment]
Yes, that's because data: is essentially the equivalent of 'unsafe-inline'.  :)

**Comment by marumari** [comment]
I will add a FAQ about it, thanks!

**Comment by tjconcept** [comment]
Embedding and image using `data:` in a CSS file compared to loading an image over https should be equally safe I assume?
How would one set the CSP to reflect that, and disallow inline scripts?

**Comment by leonklingele** [comment]
> Embedding and image using data: in a CSS file compared to loading an image over https should be equally safe I assume?

Yes

> How would one set the CSP to reflect that, and disallow inline scripts?

Set `script-src 'self'; style-src 'self'; img-src 'self' data:;`.
This will allow scripts (not inline, i.e. loaded via `<script src="..js"></script>`), allow styles (not inline, i.e. loaded via `<link href="..css" rel="stylesheet" />`) and allow images (loaded from a file and `data:` URI).

@marumari Was referring to the `data:` in the `style-src` directive.

**Comment by nico3333fr** [comment]
I had also this strange behaviour.

@marumari ```data:``` is not really ```unsafe-inline``` for styles, ```unsafe-inline``` is really more permissive for style purposes (inline-styles, etc.). Could you explain your point of view?

**Comment by marumari** [comment]
Is there any difference between allowing inline styles via data: any different than allowing them via &lt;style&gt; or style=?

If you can do this:

&lt;div style="data:text/css;base64,Ym9keSB7CiAgZm9u[...]=="&gt;, is that not the same thing as an inline style?

I might be completely misunderstanding this, as CSS related stuff is definitely not my wheelhouse.

**Comment by marumari** [comment]
(updated my comment because I was tired and silly)
